Sources/TUSKit/TUSClient.swift

```diff
@@ -547,11 +557,22 @@ func taskFor(metaData: UploadMetadata, api: TUSAPI, files: Files, chunkSize: Int
         let statusTask = StatusTask(api: api, remoteDestination: remoteDestination, metaData: metaData, files: files, chunkSize: chunkSize)
         statusTask.progressDelegate = progressDelegate
         return statusTask
-    } else {
-        let creationTask = try CreationTask(metaData: metaData, api: api, files: files, chunkSize: chunkSize)
-        creationTask.progressDelegate = progressDelegate
-        return creationTask
     }
+    
+    if let supportedExtensions {
```

Comment:
It's not entirely obvious to me why we wouldn't be leveraging the metadata's supported extensions here